src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -1395,13 +1382,14 @@ private static Encoding CreateDefaultEncoding()
         }
 
         // Returns an encoding for the system's current ANSI code page.
-        //
 
-        public static Encoding Default {
-            [System.Security.SecuritySafeCritical]  // auto-generated
-            get {
-                if (defaultEncoding == null) {
-                    defaultEncoding = CreateDefaultEncoding();
+        public static Encoding Default
+        {
+            get
+            {
+                if (defaultEncoding == null)
+                {
+                    Interlocked.CompareExchange(ref defaultEncoding, CreateDefaultEncoding(), null);
```

Comment:
The initialization of defaultEncoding can be moved to CreateDefaultEncoding. It will make it more likely that the property gets inlined.

```
        get
        {
                return defaultEncoding ?? InitializeDefaultEncoding();
        }
```